lib/Basic/LangOptions.cpp

```diff
@@ -192,6 +192,8 @@ std::pair<bool, bool> LangOptions::setTarget(llvm::Triple triple) {
     addPlatformConditionValue(PlatformConditionKind::OS, "PS4");
   else if (Target.isOSHaiku())
     addPlatformConditionValue(PlatformConditionKind::OS, "Haiku");
+  else if (Target.isOSBinFormatWasm())
```

Comment:
I don't think that we should use the `OSBinFormat` check here.  The problem is that `WASM` supports both ELF and WASM as object file formats, so if someone does `wasm-unknown-none-elf` that is going to give an error.  I think that we can check if the "architecture" is wasm, and use that to indicate that we are targeting WASM.

Reply:
[Not anymore](https://reviews.llvm.org/D48744#change-5aShrucsnrdy).

But I agree that the 'os' platform condition shouldn't be determined by the file format. If we're going to present wasm as an OS, I think it should be a kind of "default" we use when the triple's arch is wasm32/64 and the OS & vendor are both `unknown`. It's not exactly the same, but IMO that would be closer to [what clang does](https://github.com/apple/swift-clang/blob/c1ff7aaac57a47d4064268adc6d0e2ccd949c41b/lib/Basic/Targets.cpp#L566).